.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: build
+
+on: [pull_request, push, workflow_dispatch]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Pull Docker
+        run : docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      - name: Init
+        run: echo -e '#!/bin/bash\nmkdir build && cd build && cmake .. && make -Werror -Wall -Wmissing-prototypes -Wimplicit-fallthrough=3 -Wpointer-arith' > build.sh && chmod +x build.sh
+      - run: docker run -i -t -v $(pwd):/home/sof/work/sof.git --user $(id -u) sof ./build.sh
```

Comment:
> build failed: the input device is not a TTY

Please remove docker entirely here. The docker image is useful for the cross-compilation toolchain, otherwise it's pointless. If we can't easily compile rimage on any random Linux distribution then we have a serious problem.